pkg/logql/engine.go

```diff
@@ -444,11 +443,6 @@ func (q *query) JoinSampleVector(next bool, r StepResult, stepEvaluator StepEval
 		return vec, nil
 	}
 
-	stepCount := int(math.Ceil(float64(q.params.End().Sub(q.params.Start()).Nanoseconds()) / float64(q.params.Step().Nanoseconds())))
```

Comment:
Note to reviewers: the linter complained this in ineffectual assignment and I think it's right.

Reply:
I also do not see any usage of this variable